lang/test/org/partiql/lang/eval/visitors/AggregateSupportVisitorTransformTests.kt

```diff
@@ -0,0 +1,286 @@
+
+    /**
+     * Simple helper for testing that parses [this] SFW query, transforms it using [AggregateSupportVisitorTransform],
+     * and returns the transformed query as [PartiqlAst.Expr.Select].
+     */
+    private fun String.parseAndTransformQuery() : PartiqlAst.Expr.Select {
+        val query = this
+        val statement = super.parser.parseExprNode(query).toAstStatement()
+        val transformedNode = (transformer).transformStatement(statement) as PartiqlAst.Statement.Query
+        return (transformedNode.expr) as PartiqlAst.Expr.Select
+    }
+
+    /**
+     * Simple helper for testing to create a [MetaContainer] with [AggregateCallSiteListMeta] mapping to a list of
+     * [PartiqlAst.Expr.CallAgg]s.
+     *
+     * Each of the created [PartiqlAst.Expr.CallAgg]s uses the function name provided by [callAggs]'s first argument
+     * and [AggregateRegisterIdMeta] passed as the second argument of the pair.
+     */
+    private fun createCallAggMetas(callAggs: List<Pair<String, Int>>): MetaContainer =
+        metaContainerOf(AggregateCallSiteListMeta.TAG to AggregateCallSiteListMeta(
+            callAggs.map { callAgg ->
+                PartiqlAst.build {
+                    callAgg(
+                        setq = all(),
+                        funcName = callAgg.first,
+                        arg = lit(ionInt(1)),
+                        metas = metaContainerOf(AggregateRegisterIdMeta.TAG to AggregateRegisterIdMeta(callAgg.second)))
+                }
+            }))
+
+    /**
+     * Simple helper for testing to remove the [SourceLocationMeta] from [this] [MetaContainer].
+     */
+    private fun MetaContainer.removeSourceLocation() : MetaContainer = this.minus(SourceLocationMeta.TAG)
+
+    /**
+     * Checks that [expected] and [actual] have the same metas (i.e. [AggregateCallSiteListMeta])
+     * (excluding source location).
+     */
+    private fun assertSameCallAggMetas(expected: MetaContainer, actual: MetaContainer) {
+        val expectedAggCalls = (expected[AggregateCallSiteListMeta.TAG] as AggregateCallSiteListMeta).aggregateCallSites
+        val actualAggCalls = (actual[AggregateCallSiteListMeta.TAG] as AggregateCallSiteListMeta).aggregateCallSites
+        expectedAggCalls.zip(actualAggCalls).forEach { (e, a) ->
+            assertEquals(e, a)
+
+            // check that these metas have the same [AggregateRegisterIdMeta]
+            assertEquals(e.metas, a.metas.removeSourceLocation())
+        }
+    }
+
+    @Test
+    fun `one aggregate transform`() {
+        val query = "SELECT COUNT(1) FROM foo"
+        val select = query.parseAndTransformQuery()
+
+        val actualMetas = select.metas
+        val expectedMetas = createCallAggMetas(listOf(Pair("count", 0)))
+
+        assertSameCallAggMetas(expectedMetas, actualMetas)
+    }
+
+    @Test
+    fun `one aggregate in HAVING transform`() {
+        val query = "SELECT 1 FROM foo GROUP BY bar HAVING SUM(1) > 0"
```

Comment:
I think there's an opportunity to parameterize these tests here since at least the first several seem nearly identical.

Reply:
I parameterized the non-subquery tests in the latest commit. At the moment can't think of a clean way to parameterize tests (or remove some repetition) with subqueries.